src/streamlink/plugins/cnews.py

```diff
@@ -1,25 +1,43 @@
 import re
 
 from streamlink.plugin import Plugin
-from streamlink.plugin.api import useragents
+from streamlink.plugin.api import validate
+from streamlink.utils import parse_json
 
 
 class CNEWS(Plugin):
     _url_re = re.compile(r'https?://www.cnews.fr/[^ ]+')
-    _embed_video_url_re = re.compile(r'class="dm-video-embed_video" src="(?P<dm_url>.*)"')
-    _embed_live_url_re = re.compile(r'class="wrapper-live-player main-live-player"><iframe src="(?P<dm_url>.*)"')
+    _json_data_re = re.compile(r'({"basePath":.*})\);')
+    _dailymotion_url = 'https://www.dailymotion.com/embed/video/{}'
+
+    _data_schema = validate.Schema(
+        validate.any(None, validate.all(
```

Comment:
The outer `validate.any(None, validate.all(` calls are unnecessary. `validate.Schema` is already a `validate.all` wrapper.

If the response is indeed `null`, then your code will raise a `TypeError`, so I don't think you'll need the `validate.any` call.